File: _CM2022-2023ii/t23_connectivity/t23_01_e4000.py
class Graph:

    def __init__(self, adjacency_matrix):
        self.matrix = adjacency_matrix
        self.n = len(adjacency_matrix)

    def vertex_count(self, start):
        visited = {start}
        stack = [start]
        while stack:
            i = stack.pop()
            # print(i + 1)
            for j in range(self.n):
                if matrix[i][j] == 1 and j not in visited:
                    stack.append(j)
                    visited.add(j)
        return len(visited)


if __name__ == "__main__":
    with open("input.txt") as f:
        n, s = map(int, f.readline().split())
        matrix = []
        for _ in range(n):
            matrix.append(
                [int(a) for a in f.readline().split()]
            )
        graph = Graph(matrix)
        print(graph.vertex_count(s - 1))

File: _CM2024-2025ii/t23_connectivity/t23_01_e4000.php
<?php

function dfs($graph, $n, $start) {
    $count = 0;

    $visited = array_fill(0, $n, 0);
    $visited[$start] = 1;

    $stack = [];
    array_push($stack, $start);

    while (count($stack) > 0) {
        $i = array_pop($stack);
        $count++;
        for ($j = 0; $j < $n; $j++) {
            if ($graph[$i][$j] && !$visited[$j]) {
                array_push($stack, $j);
                $visited[$j] = 1;
            }
        }
    }
    return $count;
}


$lines = file('input.txt');

$data = explode(' ', $lines[0]);
$n = intval($data[0]);
$s = intval($data[1]);

$matrix = [];
for ($i = 1; $i < count($lines); $i++) {
    $row = explode(' ', $lines[$i]);
    for ($j = 0; $j < count($row); $j++) {
        $row[$j] = intval($row[$j]);
    }
    array_push($matrix, $row);
}

echo dfs($matrix, $n, $s - 1);

?>